src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -149,19 +162,116 @@ public int getTotalMessageCount(){
     return results.countEntities(FetchOptions.Builder.withLimit(1000));
   }
   
-  /* Returns all laptops */
-  public List<Laptop> getAllLaptops(){
-	 List<Laptop> laptops = new ArrayList<>();
-	 laptops.add(new Laptop("Dell","Black","Windows",15,800.00));
-	 laptops.add(new Laptop("Thinkpad","Black","Windows",13,1200.00));
-	 laptops.add(new Laptop("Dell","Grey","Windows",15,800.00));
-	 laptops.add(new Laptop("Dell","Black","Linux",15,800.00));
-	 laptops.add(new Laptop("Thinkpad","Black","Windows",15,1699.00));
-	 laptops.add(new Laptop("Macbook Pro","Black","Mac",13,1599.00));
-	 laptops.add(new Laptop("MacBook Pro","Black","Mac",15,1900.00));
-	 laptops.add(new Laptop("Macbook Pro","Grey","Mac",13,1599.00));
-	 laptops.add(new Laptop("MacBook Pro","Grey","Mac",15,1900.00));
-
-	 return laptops;
-  }
-}
\ No newline at end of file
+  /**
+   * @return list of all Laptops
+   */
+  public List<Laptop> getAlllaptopList(){
+	 return laptopList;
+  }
+  
+  /** Adds a laptop to the LaptopList
+   * @param Laptop to add to the list 
+   */
+  public void addLaptop(Laptop newLaptop) {
+	  laptopList.add(newLaptop);
+  }
+  
+  /**Returns a laptop list with laptops of a certain brand*/
+  public List<Laptop> searchBrand(String brand){
+	  //list of laptops that meet the brand requirements 
+	  List<Laptop> brandreturn=new ArrayList<>();
+	  Iterator<Laptop> it= laptopList.iterator(); 
+	  while(it.hasNext()) {
+		  if(it.next().getBrand()==brand.toLowerCase())
+			  brandreturn.add(it.next());
```

Comment:
You're calling `next()` twice in a row, which will actually iterate twice. For example, let's say I have `A`, `B`, and `C` in my list.

Step 1: I call `it.hasNext()` which returns `true`.
Step 2: I call `it.next()` which returns `A` and iterates to `B`. (I then use `A` to check the brand.)
Step 3: I call `it.next()` again which returns `B` and iterates to `C`. (I then add B to my list, even though A was the brand I checked.)

This could lead to logic problems, or it could even lead to errors: what would happen if the list only had a single element?

To fix this problem, you probbaly want to store the result of calling `it.next()` in a temporary variable, or you could use an enhanced `for` loop instead of an explicit iterator.

Reply:
(Same thing in some of the functions below.)